x-pack/plugins/security_solution/public/resolver/view/use_camera.ts

```diff
@@ -284,14 +284,30 @@ export function useAutoUpdatingClientRect(): [DOMRect | null, (node: Element | n
   const [rect, setRect] = useState<DOMRect | null>(null);
   // Using state as ref.current update does not trigger effect hook when reset
   const [currentNode, setCurrentNode] = useState<Element | null>(null);
-
+  const [currentY, setCurrentY] = useState<number>(0);
+  const [currentX, setCurrentX] = useState<number>(0);
   const ref = useCallback((node: Element | null) => {
     setCurrentNode(node);
     if (node !== null) {
       setRect(node.getBoundingClientRect());
     }
   }, []);
   const { ResizeObserver } = useContext(SideEffectContext);
+  const handleScroll = useCallback(() => {
+    if (currentX !== window.scrollX || currentY !== window.scrollY) {
```

Comment:
:red_circle: _might_ want to wrap this in a _requestAnimationFrame_ b/c I think when you query `scrollX`, `scrollY`, and `getBoundingClientRect` it forces layout sync

Reply:
re: `scrollX`, `scrollY`, `getBoundingClientRect` :

https://gist.github.com/paulirish/5d52fb081b3570c81e3a

Reply:
Is it possible `useCallback` already does this for you under the hood? Seems like something @oatkiller would know.

Reply:
@bkimmel can you explain how running that in raf would help?

Reply:
Reading `scrollX`, `scrollY` and `getBoundingClientRect` all force sync layout according to: https://gist.github.com/paulirish/5d52fb081b3570c81e3a Since while this is scrolling, it will get called rapidly, wrapping it in a rAF may reduce layout thrashing. Otherwise it seems like we'd be in a tight loop of this code forcing layout calculations and changing them in quick succession. The `rAF` doesn't solve everything, but it at least aligns the queries on scrollX etc. until after the current layout has been calculated... so I'd expect it to be at least a _little_ better. Is there a reason you know of or something I'm missing that would make it OK in this case?